### PR TITLE
Produce, consume and format TAP output

### PR DESCRIPTION
**Pull request by telphan**
Change-type: patch
Signed-off-by: Theodor Gherzan <theodor@resin.io>

### CODE REVIEW

File: entry.sh [hunk 1]
```diff
@@ -16,4 +18,5 @@ RESINOS_TESTS_RESINOS_VERSION_UPDATE=$(
 )
 
 echo "Running tests..."
-npm start
+npm  start --silent | tee $RESINOS_TESTS_RESULTS_PATH
```
Comment (jviotti) [inline entry.sh]: Looks like there are two spaces between `npm` and `start`

File: scripts/format.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const fs = require('fs')
+const summarize = require('tap-markdown')
+
+exports.mdFormatResults = (resultsFile) => {
```
Comment (jviotti) [inline scripts/format.js]: What does `md` stand for?

Comment (jviotti) [reply scripts/format.js]: Ah, its markdown. Maybe we can say `markdownFormatResults` instead, to reduce some cognitive load? :D

File: scripts/format.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  fs.createReadStream(resultsFile)
+    .pipe(summarize())
+    .pipe(fs.createWriteStream(`${resultsFile}.md`))
+  return `${resultsFile}.md`
```
Comment (jviotti) [inline scripts/format.js]: Keep in mind that streams are asynchronous, so by the time you return the file name, the file will probably not be written yet. Write streams emit a certain event (I don't remember from the top of my head, but check the docs) that you can check to make sure the write is complete.

File: scripts/format.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * limitations under the License.
+ */
+
+'use strict'
+
+const fs = require('fs')
+const summarize = require('tap-markdown')
+
+exports.mdFormatResults = (resultsFile) => {
+  fs.createReadStream(resultsFile)
```
Comment (jviotti) [inline scripts/format.js]: The read, write, (and maybe the summarize?) streams may emit `error` events that we should be handling.

File: scripts/format.js (hunk 2 above)
Comment (jviotti) [inline scripts/format.js]: Can you paste an example of how the output looks like?

**Comment by jviotti** [comment]
Also, did you test the TAP output along with the semi-manual tests? Do they still print correctly?

**Comment by telphan** [comment]
@jviotti yes, they seem to do fine.

File: scripts/format.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2017 resin.io
```
Comment (jviotti) [inline scripts/format.js]: 2018? :P

Comment (telphan) [reply scripts/format.js]: what about my heard?

File: scripts/format.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const _ = require('lodash')
+const fs = require('fs')
+const Parser = require('tap-parser')
+const Duplex = require('stream').Duplex
```
Comment (CameronDiver) [inline scripts/format.js]: I think this would be nicer as

Comment (CameronDiver) [inline scripts/format.js]: I think this would be nicer as

Comment (CameronDiver) [inline scripts/format.js]: I think this would be nicer as 
```
const { Duplex } = require('stream');
```

File: scripts/format.js [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+const symbols = require('figures')
+const indentString = require('indent-string')
+
+const LF = '\n'
```
Comment (CameronDiver) [inline scripts/format.js]: This should be `const LF = require('os').EOL;`

File: scripts/format.js [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+const wrapParser = (tap) => {
+  const dup = Duplex()
+
+  // eslint-disable-next-line no-underscore-dangle
+  dup._read = () => {
```
Comment (CameronDiver) [inline scripts/format.js]: This can be `dup._read = _.noop`;

**Pull request merged**